### PR TITLE
Add bottlerocket admin and control images, custom host and bootstrap containers support

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

https://github.com/aws/eks-anywhere/issues/3283

*Description of changes:*

To provision unstacked Etcd node on Bottlerocket OS for snow, we need override the default admin, control image for BR AWS variant. Also need to pass in the Snow BR bootstrap container setting similar to CAPI: https://github.com/aws/eks-anywhere-build-tooling/pull/1599.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: api/v1beta1/etcdadmconfig_types.go [hunk 1]
```diff
@@ -97,8 +97,64 @@ type BottlerocketConfig struct {
...
+	// CustomBootstrapContainers adds additional bootstrap containers for bottlerocket.
+	// +optional
+	CustomBootstrapContainers []BottlerocketBootstrapContainer `json:"customBootstrapContainers,omitempty"`
+}
+
+// BottlerocketHostContainer holds the host container setting for bottlerocket.
+type BottlerocketHostContainer struct {
+	// Name is the host container name that will be given to the container in BR's `apiserver`
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
```
Comment (vivek-koppuru) [inline api/v1beta1/etcdadmconfig_types.go:122]: Name, Image, and UserData can technically be shared between the two structs as a separate one that is inline right? It will help us avoid having to change those fields in two places

File: api/v1beta1/etcdadmconfig_types.go [hunk 2]
```diff
@@ -97,8 +97,64 @@ type BottlerocketConfig struct {
...
+
+	// Image is the actual image used for Bottlerocket bootstrap.
+	Image string `json:"image"`
+
+	// Essential decides whether or not the container should fail the boot process.
+	// Bootstrap containers configured with essential = true will stop the boot process if they exit code is a non-zero value.
+	// Default is false.
+	// +optional
+	Essential bool `json:"essential"`
```
Comment (vivek-koppuru) [inline api/v1beta1/etcdadmconfig_types.go:149]: Superpowered and Essential are terms used in BR?

File: pkg/userdata/bottlerocket/node_userdata.go [hunk 3]
```diff
@@ -14,12 +14,36 @@ import (
 )
 
 const (
-	adminContainerInitTemplate = `{{ define "adminContainerInitSettings" -}}
```
Comment (vivek-koppuru) [inline pkg/userdata/bottlerocket/node_userdata.go:17]: What's happening to the admin container then? I see that we are introducing a field `AdminImage` above so how is that related to here.

File: pkg/userdata/bottlerocket/node_userdata.go [hunk 4]
```diff
@@ -106,11 +118,33 @@ func generateBottlerocketNodeUserData(bootstrapContainerUserData []byte, users [
 	}
 	b64AdminContainerUserData := base64.StdEncoding.EncodeToString(adminContainerUserData)
 
+	hostContainers := []etcdbootstrapv1.BottlerocketHostContainer{
+		{
+			Name:         "admin",
+			Superpowered: true,
+			Image:        config.BottlerocketConfig.AdminImage,
```
Comment (vivek-koppuru) [inline pkg/userdata/bottlerocket/node_userdata.go:125]: Oh I see, it is being decoupled here. Which is why we need AdminImage at a higher level separately now?

**Review by vivek-koppuru — APPROVED** [review]

File: api/v1beta1/etcdadmconfig_types.go (hunk 2 above)
Comment (jiayiwang7) [reply api/v1beta1/etcdadmconfig_types.go:149]: yup

File: pkg/userdata/bottlerocket/node_userdata.go (hunk 3 above)
Comment (jiayiwang7) [reply pkg/userdata/bottlerocket/node_userdata.go:17]: admin container is part of the hostContainers. We now loop through the host containers (admin, control, kubeadm-bootstrap) and generate the template

File: pkg/userdata/bottlerocket/node_userdata.go (hunk 4 above)
Comment (jiayiwang7) [reply pkg/userdata/bottlerocket/node_userdata.go:125]: yup

File: api/v1beta1/etcdadmconfig_types.go (hunk 1 above)
Comment (jiayiwang7) [reply api/v1beta1/etcdadmconfig_types.go:122]: yea.. i don't feel strongly abt combining them just bc they share same fields - besides this is consistent w the capi BR fields. i'd prefer to leave it as it is unless we feel necessary to change it
